internal/resource/resource.go

```diff
@@ -45,10 +47,12 @@ func ReadAndDecodeResourceFile(resourceFile string) (*unstructured.Unstructured,
 	return &u, nil
 }
 
-func GetResourceType(resourceFile string) (string, error) {
-	u, err := ReadAndDecodeResourceFile(resourceFile)
-	if err != nil {
-		return "", err
+func GetResourceType(filename string) (string, error) {
+	if strings.Contains(filename, "step-action") || strings.Contains(filename, "stepactions") {
+		return "StepAction", nil
+	} else if strings.Contains(filename, "task") || strings.Contains(filename, "tasks") {
+		return "Task", nil
```

Comment:
I am not sure why this changed ? 🤔 It should work just as well with `StepAction` doesn't it ?
And if it has to change, isn't it missing the `Pipeline` type ?

Reply:
@vdemeester Actually, I wanted the tasks and stepActions to be well segregated into separate folders. With these code changes in place the catalog.yaml looks somewhat likes this :- 

`version: v1
catalog:
  repository:
    description: ""
  attestation:
    publickey: ""
  resources:
    tasks:
      - name: git-cli
        version: 0.5.5
        filename: tasks/task-git-cli/task-git-cli.yaml
        checksum: 3f6d75b0757ef050d204c8fd19e13b2491c788959a1f7eed652af59a5266a6d2
        signature: ""
      - name: git-clone
        version: 0.5.5
        filename: tasks/task-git-clone/task-git-clone.yaml
        checksum: 6ef249e79bd644b55035a45794fa499fa51dbbfa0cc8a4c0e5c92111a8e0533c
        signature: ""
    pipelines: []
    stepactions:
      - name: git-clone
        version: 0.5.5
        filename: stepactions/step-action-git-clone/step-action-git-clone.yaml
        checksum: c624d479f251a6dc145f50bcc85945657cc6194a6b2b0cf6c9976467f55f9cf9
        signature: ""`

Reply:
> ok 🤔 But it would have look the same without this change, isn't it ? (and if you have a pipeline as well, doesn't it fail ?)

No no they look different @vdemeester . I have shared the difference in slack you can check.

Reply:
> ok 🤔 But it would have look the same without this change, isn't it ? (and if you have a pipeline as well, doesn't it fail ?)

Yes those changes are not required and I have reverted back to the original code.